storage/sequence_cache.go

```diff
@@ -43,73 +44,96 @@ var errEmptyID = errors.New("empty CommandID used in sequence cache")
 // A SequenceCache is not thread safe. Access to it is serialized
 // through Raft.
 type SequenceCache struct {
-	rangeID roachpb.RangeID
+	rangeID      roachpb.RangeID
+	min, max     roachpb.Key
+	scratchEntry roachpb.SequenceCacheEntry
+	scratchBuf   [256]byte
 }
 
 // NewSequenceCache returns a new sequence cache. Every range replica
 // maintains a sequence cache, not just the leader.
 func NewSequenceCache(rangeID roachpb.RangeID) *SequenceCache {
+	// The sequence number is encoded in decreasing order. We could get away
+	// with zero due to the fact that UUIDs always contain a '4', but let's
+	// not.
+	seqCacheKeyMin, _ := keys.SequenceCacheKey(rangeID, uuidMin, math.MaxUint32)
+	seqCacheKeyMax, _ := keys.SequenceCacheKey(rangeID, uuidMax, 0)
+
 	return &SequenceCache{
 		rangeID: rangeID,
+		min:     seqCacheKeyMin,
+		max:     seqCacheKeyMax,
 	}
 }
 
-// ClearData removes all persisted items stored in the persistent.
-func (rc *SequenceCache) ClearData(e engine.Engine) error {
-	from := keys.SequenceCacheKey(rc.rangeID, roachpb.KeyMin)
-	to := keys.SequenceCacheKey(rc.rangeID, roachpb.KeyMax)
-	_, err := engine.ClearRange(e, engine.MVCCEncodeKey(from), engine.MVCCEncodeKey(to))
+var uuidMin = bytes.Repeat([]byte{'\x00'}, roachpb.TransactionIDLen)
```

Comment:
These variables would be better named `transactionID{Min,Max}` (or `TransactionIDLen` renamed to `UUIDLen`)